code/lightna/magento-os-frontend/template/page/header/minicart.phtml

```diff
@@ -1,6 +1,57 @@
 <?php
 /** @var Lightna\Magento\Data\Session $session */
+/** @var Lightna\Magento\Data\Currency $currency */
+$cart = $session->cart;
 ?>
-<div class="cjs-minicart icon cart cursor-pointer">
-    <div class="cjs-minicart-counter counter"><?= $session->cart->qty() ?></div>
+<div class="cjs-minicart">
+    <div class="icon cart cursor-pointer" data-action="open-minicart">
+        <div class="counter"><?= $cart->qty() ?></div>
+    </div>
+<?php if ($_SERVER['REQUEST_METHOD'] === 'POST') { ?>
```

Comment:
I liked the split for content as separate template but name wasn't good. minicart/content coul be better. Why you removed it?

Reply:
so counter and content in the same template, or I misunderstood your idea?

Reply:
ah, confusion seems in "same template", I meant same block which doesn't affects templates structure, sorry for confusion

Reply:
could you also change the condition, now it can be @var Lightna\Engine\Data\Request $request
if(!$request->isPost)

Reply:
but you need to pull

Reply:
still not sure we're on the same page, have another look please 😃

Reply:
short call wanted

Reply:
replaced template with block; split content template on several ones